core/src/test/scala/org/apache/spark/rdd/PairRDDFunctionsSuite.scala

```diff
@@ -864,6 +872,47 @@ class NewFakeFormatWithCallback() extends NewFakeFormat {
   }
 }
 
+class YetAnotherFakeCommitter extends NewOutputCommitter with Assertions {
+  def setupJob(j: NewJobContext): Unit = {
+    JobID.jobid = j.getJobID().getId
+    ()
+  }
+
+  def needsTaskCommit(t: NewTaskAttempContext): Boolean = false
+
+  def setupTask(t: NewTaskAttempContext): Unit = {
+    val rddid = t.getTaskAttemptID().getJobID().getId
+    assert(rddid === RDDID.rddid )
+    assert(rddid === JobID.jobid)
+    ()
+  }
+
+  def commitTask(t: NewTaskAttempContext): Unit = ()
+
+  def abortTask(t: NewTaskAttempContext): Unit = ()
```

Comment:
our convention is to put empty *braces* `{ }` not empty parens for no-op methods.